app/src/main/java/org/mozilla/fenix/utils/WifiConnectionMonitor.kt

```diff
@@ -0,0 +1,77 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.utils
+
+import android.app.Application
+import android.content.Context
+import android.net.ConnectivityManager
+import android.net.Network
+import android.net.NetworkCapabilities
+import android.net.NetworkRequest
+
+/**
+ * Attaches itself to the [Application] and listens for WIFI available/not available events. This
+ * allows calling code to set simpler listeners.
+ *
+ * Example:
+ * ```kotlin
+ *  app.components.wifiConnectionListener.addOnWifiConnectedChangedListener { isConnected ->
+ *    if (isConnected) {
+ *      downloadThing()
+ *    }
+ *  }
+ * ```
+ */
+class WifiConnectionMonitor(app: Application) {
+    private val callbacks = mutableSetOf<(Boolean) -> Unit>()
+
+    private var lastKnownStateWasAvailable = false
+
+    init {
+        object : ConnectivityManager.NetworkCallback() {
+            init {
+                val request = NetworkRequest.Builder()
+                    .addTransportType(NetworkCapabilities.TRANSPORT_WIFI)
+                    .build()
+
+                val cm = app.getSystemService(Context.CONNECTIVITY_SERVICE) as
+                        ConnectivityManager
+
+                cm.registerNetworkCallback(request, this)
```

Comment:
Should this class have `start` and `stop` function this way consumers can control when to start/stop listing i.e. ([registerNetworkCallback](https://developer.android.com/reference/android/net/ConnectivityManager#unregisterNetworkCallback(android.net.ConnectivityManager.NetworkCallback)))?

Reply:
I could go either way on this one.  We could always add it when we encountered a use case.  I like that setting the wifi listener is a simple as just saying `monitor.addOnWifiConnectedChangedListener { }`.

Reply:
My only concern is performance, we are only interested on calls when the app is in the foreground, otherwise will be getting notified without any gain. As we just want to consume what is strictly needed Wdy? :)

Reply:
I don't expect this to block anything important, so I'm not massively worried about it, but it's still a reasonable concern.  Sure, I'll add it in.